tests/scripts/test-e2e-cluster.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash -e
+# Input env variables:
+# - PROVIDER - the LLM provider to be used during the test
+# - PROVIDER_KEY_PATH - path to a file containing the credentials to be used with the llm provider
+# - MODEL - name of the model to use during e2e testing
+
+# Script flow:
+# 1) Install OLS into a namespace on the cluster with valid config/api tokens and exposed via a route
+# 2) Setup a service account w/ permission to access OLS
+# 3) Wait for the ols api server to be available
+# 4) Invoke the test-e2e Makefile target
+
+
+DIR="${BASH_SOURCE%/*}"
+if [[ ! -d "$DIR" ]]; then DIR="$PWD"; fi
+. "$DIR/utils.sh"
+
+# temp directory for generated resource yamls
+TMPDIR=$(mktemp -d)
+
+# ARTIFACT_DIR is defined when running in a prow job, content
+# in this location is automatically collected at the end of the test job
+ARTIFACT_DIR=${ARTIFACT_DIR:-$TMPDIR}
+
+# Deletes may fail if this is the first time running against
+# the cluster, so ignore failures
+oc delete ns openshift-lightspeed || true
+oc delete clusterrole ols-sar-check || true
+oc delete clusterrolebinding ols-sar-check || true
+oc delete clusterrole ols-user || true
+
+oc create ns openshift-lightspeed
+oc project openshift-lightspeed
+
+# create the llm api key secret ols will mount
+oc create secret generic llmcreds --from-file=llmkey="$PROVIDER_KEY_PATH"
+
+# create the configmap containing the ols config yaml
+envsubst < tests/config/cluster_install/ols_configmap.yaml > "$ARTIFACT_DIR/ols_configmap.yaml"
+oc create -f "$ARTIFACT_DIR/ols_configmap.yaml"
+
+# create the ols deployment and related resources (service, route, rbac roles)
+envsubst < tests/config/cluster_install/ols_manifests.yaml > "$ARTIFACT_DIR/ols_manifests.yaml"
+oc create -f "$ARTIFACT_DIR/ols_manifests.yaml"
+
+# create a new service account with no special permissions and get an auth token for it
+oc create sa olsuser
```

Comment:
I think this is calling for a
```
oc wait --timeout=120s --for=condition=available sa/olsuser
```
afterwards or we're going to have a race with the uses of sa/olsuser below.

Reply:
service accounts don't have conditions, so i don't think you can do this?  we'd need to use the jsonpath expression check.

i haven't seen any races in my testing, either because the creation is atomic, or because you can create rolebindings that reference users that don't necessarily exist.

I was slightly worried about a race getting the token for the olsuser SA, but so far i haven't seen it being a problem.

Reply:
Whatever races there are, they will eventually surface.